pkg/resource/deploy/source_eval.go

```diff
@@ -728,7 +728,6 @@ func (rm *resmon) RegisterResource(ctx context.Context,
 	}
 
 	state := result.State
-	props = state.All()
```

Comment:
This removal does not seem to be correct by itself - it leads to `RegisterResource` returning inputs instead of outputs. 

The re-use of `props` to now mean outputs starting here is extremely confusing though.  I believe `state.Outputs` is the value that should be used in place of `props` below.

Reply:
Should be addressed with https://github.com/pulumi/pulumi/pull/2659/commits/83749fa9b7ee0403a828fc22b0742c4125cd9746.

Reply:
Lgtm.